Sources/ComposableArchitecture/ViewStore.swift

```diff
@@ -79,7 +77,7 @@ public final class ViewStore<State, Action>: ObservableObject {
       instrumentation.callback?(sendCallbackInfo, .pre, .viewStoreSend)
       defer { instrumentation.callback?(sendCallbackInfo, .post, .viewStoreSend) }
 
-      store.send($0, file: file, line: line, instrumentation: instrumentation)
+      return store.send($0, file: file, line: line, instrumentation: instrumentation)
```

Comment:
If the resulting `Task` is not being awaited here will the `defer` be executed potentially before the `Task`?

Reply:
`send` still runs sync in this context, but only returns a `Task`, which can be awaited on. All the sync actions have been processed at the return time of the `send`, but the task can be used to cancel (or wait for) any long running effects resulting from this action.